### ISSUE TITLE
CI estimation problem for small internal noise

### ISSUE THREAD

**Issue by jjau** [neuro-team-femto/palin#15]
# What

GLMMethod estimates confidence interval width based on GLMKernel. Small internal noise <1 appear to generate estimation error, where the CI increases as internal noise decreases. This is somewhat corrected by adding jitter, but this is not satisfying. 

![Image](https://github.com/user-attachments/assets/299032e6-875d-4745-9d32-225f6300f002)

# Todo

Test alternative GLM training methods (ex. other python models, R functions) and compare. 

# Test

Something like: 

```
observer_params = {'kernel':['random'],#[[0,0,0,0,0]]
                   'internal_noise_std':np.arange(0,5.1,0.1), 
                  'criteria':[0]}
experiment_params = {'n_trials':[100,200, 500,1000, 5000], 
                     #'n_repeated':[1000],
                     'trial_type': [Int2Trial],
                     'n_features': [5],
                     'external_noise_std': [100]}
analyser_params = {'agg_mode':['argmax'], 
                  'jitter':[.0001, .001, .005, .01, .02, .03, .05, .1, .2, .5]}        
                   
sim = Sim(SimpleExperiment, experiment_params, 
          LinearObserver, observer_params, 
          CIValues, analyser_params)
sim_df = sim.run_all(n_runs=5)

sns.lineplot(data=sim_df[(sim_df.agg_mode=='argmax')&(sim_df.jitter<.2)], x='internal_noise_std', y='confidence_interval', hue='jitter')
```

**Comment by jjau** [comment]
ex. logit regression de scikit

**Comment by lnalborczyk** [comment]
In case the `scikit` option does not work out, the `R` method can be called via `palin::computing_kernel(data = some_data, method = "glm")`, which returns the kernel (computed via a GLM) and the confidence interval around the kernel.

**Comment by aynazadl** [comment]
I looked at all 3 options: 1: statsmodels :: smf.glm (logit & probit)& smf.logit , can not handle internal_noise==0, scikit-learn and rpy2(stats :: glm) can handle 0 noise but scikit-learn doesn't give directly the confidence interval.

**Comment by aynazadl** [comment]
it's not obvious that rpy2 doesn't have the same small-IN problem (it's actually bigger here, because python methods use jitter to mitigate it, and rpy2 don't), so perhaps we should just use python methods ? (knowing that requiring rpy2 is not user friendly)

![Image](https://github.com/user-attachments/assets/e7f5a262-7c19-4f57-a4ab-170e71859b1f)

**Comment by lnalborczyk** [comment]
> it's not obvious that rpy2 doesn't have the same small-IN problem (it's actually bigger here, because python methods use jitter to mitigate it, and rpy2 don't), so perhaps we should just use python methods ? (knowing that requiring rpy2 is not user friendly)
> 
> ![Image](https://github.com/user-attachments/assets/e7f5a262-7c19-4f57-a4ab-170e71859b1f)

I don't obtain the same results with palin-r so I guess the discrepancy comes from the method for "normalising" the CI (and the subsequent regression for predicting the IN).
